### PR TITLE
[RFR] Implement `<Boolean />` field (follows #50)

### PR DESCRIPTION
**Pull request by jpetitcolas**
- [x] Implement `<Boolean />` input
- [x] Implement `<NotNullableBoolean />` input
- [x] Implement `<BooleanField />` component
- [x] Fix select list with falsy values material-ui issue (see https://github.com/callemall/material-ui/issues/4275)
- [x] Update documentation

### CODE REVIEW

File: example/posts.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import React from 'react';
 import {
+    BooleanField,
+    BooleanInput,
```
Comment (jpetitcolas) [inline example/posts.js]: Not used

**Comment by Kmaschta** [comment]
What about rename the inputs:
- `<NotNullableBoolean />` to `<Boolean />` (toggle button)
- `<Boolean />` to `<NullableBoolean />` (select)

It would be more clear to me.

File: src/mui/input/BooleanInput.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React, { PropTypes } from 'react';
+import SelectInput from './SelectInput';
+
+const BooleanInput = ({ input, meta: { touched, error }, label }) => (
+    <SelectInput
+        input={input}
+        label={label}
+        choices={[
+            { value: null, label: 'Unknown' }, // can't set it to null to prevent from label overlap issues
```
Comment (Kmaschta) [inline src/mui/input/BooleanInput.js]: `Undefined` would be a better label

**Comment by jpetitcolas** [comment]
Switching to RFR. 

But do not merge it yet: it requires https://github.com/callemall/material-ui/pull/5302 to be merged first.

**Comment by jpetitcolas** [comment]
(and code review applied)

File: src/mui/input/BooleanInput.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    label: {
+        color: 'rgba(0, 0, 0, 0.298039)',
+    },
+    toggle: {
+        marginBottom: 16,
+    },
+};
+
+const NotNullableBooleanInput = ({ input, label }) => (
```
Comment (Kmaschta) [inline src/mui/input/BooleanInput.js]: s/NotNullableBooleanInput/BooleanInput/

File: src/mui/input/BooleanInput.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        <Toggle
+            {...input}
+            labelStyle={styles.label}
+            style={styles.toggle}
+            label={label}
+        />
+    </div>
+);
+
+NotNullableBooleanInput.propTypes = {
```
Comment (Kmaschta) [inline src/mui/input/BooleanInput.js]: s/NotNullableBooleanInput/BooleanInput/

File: src/mui/input/BooleanInput.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    </div>
+);
+
+BooleanInput.propTypes = {
+    includesLabel: PropTypes.bool.isRequired,
+    input: PropTypes.object,
+    label: PropTypes.string,
+};
+
+NotNullableBooleanInput.defaultProps = {
```
Comment (Kmaschta) [inline src/mui/input/BooleanInput.js]: s/NotNullableBooleanInput/BooleanInput/

File: src/mui/input/BooleanInput.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    includesLabel: PropTypes.bool.isRequired,
+    input: PropTypes.object,
+    label: PropTypes.string,
+};
+
+BooleanInput.defaultProps = {
+    includesLabel: true,
+};
+
+export default NotNullableBooleanInput;
```
Comment (Kmaschta) [inline src/mui/input/BooleanInput.js]: s/NotNullableBooleanInput/BooleanInput/

File: src/mui/input/NullableBooleanInput.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import React, { PropTypes } from 'react';
+import SelectInput from './SelectInput';
+
+const BooleanInput = ({ input, meta: { touched, error }, label }) => (
```
Comment (Kmaschta) [inline src/mui/input/NullableBooleanInput.js]: s/BooleanInput/NullableBooleanInput/

File: src/mui/input/NullableBooleanInput.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+            { value: false, label: 'No' },
+            { value: true, label: 'Yes' },
+        ]}
+        optionText="label"
+        optionValue="value"
+        errorText={touched && error}
+    />
+);
+
+BooleanInput.propTypes = {
```
Comment (Kmaschta) [inline src/mui/input/NullableBooleanInput.js]: s/BooleanInput/NullableBooleanInput/

File: src/mui/input/NullableBooleanInput.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    includesLabel: PropTypes.bool.isRequired,
+    input: PropTypes.object,
+    label: PropTypes.string,
+    meta: PropTypes.object,
+};
+
+BooleanInput.defaultProps = {
```
Comment (Kmaschta) [inline src/mui/input/NullableBooleanInput.js]: s/BooleanInput/NullableBooleanInput/

File: src/mui/input/NullableBooleanInput.js [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+    includesLabel: true,
+};
+
+export default BooleanInput;
```
Comment (Kmaschta) [inline src/mui/input/NullableBooleanInput.js]: s/BooleanInput/NullableBooleanInput/

File: src/mui/input/NullableBooleanInput.js [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+            { value: false, label: 'No' },
+            { value: true, label: 'Yes' },
+        ]}
+        optionText="label"
+        optionValue="value"
+        errorText={touched && error}
+    />
+);
+
+BooleanInput.propTypes = {
```
Comment (Kmaschta) [inline src/mui/input/NullableBooleanInput.js]: s/BooleanInput/NullableBooleanInput/

File: src/mui/input/NullableBooleanInput.js [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    includesLabel: PropTypes.bool.isRequired,
+    input: PropTypes.object,
+    label: PropTypes.string,
+    meta: PropTypes.object,
+};
+
+BooleanInput.defaultProps = {
```
Comment (Kmaschta) [inline src/mui/input/NullableBooleanInput.js]: s/BooleanInput/NullableBooleanInput/

**Comment by djhi** [comment]
https://github.com/callemall/material-ui/pull/5302 has been closed as it has been fixed by another PR

**Comment by fzaninotto** [comment]
Yet we still need the fix to be released.

File: src/mui/input/BooleanInput.js [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+    },
+};
+
+const BooleanInput = ({ input, label }) => (
+    <div style={styles.block}>
+        <Toggle
+            {...input}
+            labelStyle={styles.label}
+            style={styles.toggle}
+            label={label}
```
Comment (mstmustisnt) [inline src/mui/input/BooleanInput.js]: we can add 
```
defaultToggled={input.value}
```

to make it toggled if record[source] is `true`

**Comment by ilaif** [comment]
Hey guys, great project!
Does this feature have a release date?
Thanks.

**Comment by mosasiru** [comment]
Good project.
I want to try this feature.

**Comment by fzaninotto** [comment]
I think https://github.com/callemall/material-ui/pull/5360 has been merged and released in material-ui 0.16.1, which is the current version dependency. 

@jpetitcolas can you rebase?

**Comment by fzaninotto** [comment]
Rebased, mergeable

**Pull request merged**